stream-chat-android-client/src/main/java/io/getstream/chat/android/client/plugin/Plugin.kt

```diff
@@ -274,4 +279,8 @@ public interface Plugin :
         channelId: String,
         memberIds: List<String>,
     ): Result<Unit> = Result.success(Unit)
+
+    public fun onUserSet(user: User)
```

Comment:
We are changing from `Composition` to `Inheritance` to handle user connection/disconnection. Instead of having a class to handle this matter (InitializationCoordinator), we are now passing the responsibility to the plugin itself, which becomes more complex. 

The way I see it, `StatePlugin` become a class that handles everything related to state, so a single point of access. But the problem is that it also has so many responsibilities related to the logic of state in the SDK. 

I believe a class specialized in the lifecycle of a plugin would be better in this situation. Something like PluginHandler/PluginManager/PluginLifecycle takes care of connection and disconnection, so the plugin itself doesn't have to add those methods on its signature.